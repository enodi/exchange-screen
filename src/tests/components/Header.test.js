import React from "react";
import { shallow } from "enzyme";
import Header from "components/Header";

describe("Header Component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Header />);
  });

  it("should render Header component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should contain h1 element with text", () => {
    expect(wrapper.find("h1").text()).toEqual("Revolut");
  });

  it("should contain attribute that navigates to home page", () => {
    expect(wrapper.find("NavLink").prop("to")).toEqual("/");
  });
});
