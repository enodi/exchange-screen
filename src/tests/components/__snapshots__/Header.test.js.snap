// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render Header component correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <NavLink
        className="header__title"
        to="/"
      >
        <h1>
          Revolut
        </h1>
        <img
          alt="logo"
          className="logo"
          src="bar.svg"
        />
      </NavLink>
    </div>
  </div>
</header>
`;
